### PR TITLE
fix(API): reduce code duplication between DB entities

### PR DESCRIPTION
**Pull request by netroy**

### CODE REVIEW

**Review by ivov — COMMENTED** [review]
Tested all three DBs + PG alt schema. Also tested recreating all three DBs manually.

We have to be careful in future not to extend from `AbstractEntity` if we don't need the shared fields, e.g. `Settings`

**Comment by netroy** [comment]
> We have to be careful in future not to extend from `AbstractEntity` if we don't need the shared fields, e.g. `Settings`

Should we create a separate `AbstractEntityWithTimestamps` ?

**Comment by ivov** [comment]
> Should we create a separate `AbstractEntityWithTimestamps` ?

No, that's fine. Cases like `Settings` should be rare. Just to keep in mind.

File: packages/cli/src/databases/entities/ExecutionEntity.ts [hunk 1]
```diff
@@ -47,19 +29,19 @@ export class ExecutionEntity implements IExecutionFlattedDb {
 	@Column({ nullable: true })
 	retrySuccessId: string;
 
-	@Column(resolveDataType('datetime'))
+	@Column(datetimeColumnType)
 	startedAt: Date;
 
 	@Index()
-	@Column({ type: resolveDataType('datetime') as ColumnOptions['type'], nullable: true })
+	@Column({ type: datetimeColumnType, nullable: true })
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/databases/entities/ExecutionEntity.ts:36]: `startedAt` isn't nullable

File: packages/cli/src/databases/entities/User.ts [hunk 2]
```diff
@@ -121,21 +87,10 @@ export class User implements IUser {
...
-	@UpdateDateColumn({
-		precision: 3,
-		default: () => getTimestampSyntax(),
-		onUpdate: getTimestampSyntax(),
-	})
-	updatedAt: Date;
-
 	@BeforeInsert()
 	@BeforeUpdate()
 	preUpsertHook(): void {
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/cli/src/databases/entities/User.ts:92]: looks like all `BeforeUpdate` methods get called, which is why I removed the `this.updatedAt` from here.

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with `n8n@0.200.0`

**Pull request merged**